dashboard/src/components/DeploymentForm/BasicDeploymentForm/Subsection.test.tsx

```diff
@@ -36,17 +38,19 @@ const defaultProps = {
   handleBasicFormParamChange: jest.fn(),
   appValues: "externalDatabase: {}",
   handleValuesChange: jest.fn(),
-} as IDatabaseSectionProps;
+  enablerChildrenParam: "useSelfHostedDatabase",
+  enablerCondition: false,
+} as ISubsectionProps;
 
 it("should render a external database section", () => {
-  const wrapper = shallow(<DatabaseSection {...defaultProps} />);
+  const wrapper = shallow(<Subsection {...defaultProps} />);
   expect(wrapper).toMatchSnapshot();
 });
 
 it("should hide/show the database params if the self-hosted database is enabled/disabled", () => {
-  const wrapper = shallow(<DatabaseSection {...defaultProps} />);
+  const wrapper = shallow(<Subsection {...defaultProps} />);
   expect(defaultProps.param.children!.useSelfHostedDatabase.value).toBe(true);
-  expect(wrapper.find(".margin-t-normal").prop("hidden")).toBe(true);
+  expect(wrapper.find("div").findWhere(d => d.prop("hidden"))).toExist();
```

Comment:
> This change is because there are more than one div ? So it's "expect at least one div with a hidden prop". Not sure if that should be more specific to match the test label.

Yes. Taking a look to the functional code, I couldn't find something better to identify the `div` I am interested in.

> Also - why are you creating a div with a hidden prop, rather than a className or similar?

Well, I am interested in hiding the `div`, why are you suggesting using a className instead? Is there something I am missing?

> Also, does this return true for <div hidden={false}> as the prop is still passed (which is what the code below does, I think?)

No, since the property of `hidden` is false, the find iteration return `false` so it's not returned. It only returns `true` for `<div hidden={true}>`.

Reply:
> Well, I am interested in hiding the `div`, why are you suggesting using a className instead? Is there something I am missing?

No - I was just unsure whether the `hidden` attribute should be used in this case. See the [MDN doc](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/hidden), I was thinking that what we're doing here is equivalent to hiding panels in a tabbed dialog (where it should not be used), but the use-case here is different in that the elements are only relevant if the self-hosted option is enabled (across all presentations) so I think you're right to use it.